File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/leakcanary/LeakCanaryService.java
package com.jdroid.android.sample.ui.leakcanary;

import android.content.Intent;
import androidx.annotation.NonNull;

import com.jdroid.android.service.AbstractWorkerService;

public class LeakCanaryService extends AbstractWorkerService {

	public static AbstractWorkerService LEAK;

	@Override
	protected void doExecute(@NonNull Intent intent) {
		LEAK = this;
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/leakcanary/LeakCanaryService.kt
package com.jdroid.android.sample.ui.leakcanary

import android.content.Intent

import com.jdroid.android.service.AbstractWorkerService

class LeakCanaryService : AbstractWorkerService() {

    override fun doExecute(intent: Intent) {
        LEAK = this
    }

    companion object {
        var LEAK: AbstractWorkerService? = null
    }
}
